src/hooks/use-models.ts

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useState, useStaleLayout, useTheme, useEmbed } from '@nebula.js/stardust';
+import createContainerModel from '../models/container-model';
+import { layoutService as createLayoutService, themeService as createThemeService } from 'qlik-chart-modules';
+import themeStyleMatrix from '../services/theme-service/theme-style-matrix';
+import layoutServiceMeta from '../services/layout-service/meta';
+
+type UseModelsProps = {
+  core: {
+    element: HTMLElement;
+  };
+};
+
+const UseModels = ({ core }: UseModelsProps) => {
+  const layout = useStaleLayout();
+  const theme = useTheme();
+  const embed = useEmbed();
+
+  const [models, setModels] = useState();
+
+  useEffect(() => {
+    if (!core) {
+      return;
+    }
+    const layoutService = createLayoutService({
+      source: layout,
+      metaAdditionsFn: layoutServiceMeta(),
+    });
+    const themeService = createThemeService({ theme, styleMatrix: themeStyleMatrix });
+
+    const containerModel = createContainerModel({ layoutService, themeService, embed });
+    setModels({
+      containerModel,
+    });
+  }, [core]);
```

Comment:
You will want to update the models and services when the layout and app changes.

Reply:
Done